src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java

```diff
@@ -879,6 +887,224 @@ private static File getSampleMapFile(final Map<String, String> mapping){
+
+            final Path originalVCFPath = originalVCFFile.toPath();
+            final String uncompressedVCFName = originalVCFFile.getName().replaceAll("\\.gz$", "");
+            Path vcfDestination = new File(vcfDir, originalVCFFile.getName()).toPath();
+            if ( useTribbleIndex ) {
+                vcfDestination = new File(vcfDir, uncompressedVCFName).toPath();
+                IOUtils.gunzip(originalVCFPath.toAbsolutePath().toFile(), vcfDestination.toAbsolutePath().toFile());
+            } else {
+                Files.copy(originalVCFPath, vcfDestination);
+            }
+
+            final File originalVCFIndexFile = new File(originalVCFFile.getAbsolutePath() + FileExtensions.TABIX_INDEX);
+            Assert.assertTrue(originalVCFIndexFile.exists());
+            final File thisVCFIndexDir = createExplicitIndex ? indexDir : vcfDir;
+            Path vcfIndexDestination = new File(thisVCFIndexDir, originalVCFIndexFile.getName()).toPath();
+            if ( useTribbleIndex ) {
+                vcfIndexDestination = new File(thisVCFIndexDir, uncompressedVCFName + FileExtensions.TRIBBLE_INDEX).toPath();
+                final Index inMemoryIndex = IndexFactory.createLinearIndex(vcfDestination, new VCFCodec(), IndexFeatureFile.OPTIMAL_GVCF_INDEX_BIN_SIZE);
+                inMemoryIndex.write(vcfIndexDestination);
+            } else {
+                Files.copy(originalVCFIndexFile.toPath(), vcfIndexDestination);
+            }
+
+            if ( createExplicitIndex ) {
+              sampleNameMapContents.append(String.format("%s\t%s\t%s\n", sampleName, vcfDestination.toAbsolutePath().toString(), vcfIndexDestination.toAbsolutePath().toString()));
+            } else {
+              sampleNameMapContents.append(String.format("%s\t%s\n", sampleName, vcfDestination.toAbsolutePath().toString()));
+            }
+        }
+
+        final File sampleNameMapFile = IOUtils.writeTempFile(sampleNameMapContents.toString(), "testExplicitIndicesInSampleNameMap_samplemap", ".txt");
+
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        args.add(GenomicsDBImport.SAMPLE_NAME_MAP_LONG_NAME, sampleNameMapFile.getAbsolutePath())
+                .addInterval(INTERVAL.get(0))
+                .add(GenomicsDBImport.WORKSPACE_ARG_LONG_NAME, workspace);
+        runCommandLine(args);
+
+        checkJSONFilesAreWritten(workspace);
+        checkGenomicsDBAgainstExpected(workspace, INTERVAL, COMBINED, b38_reference_20_21, true, ATTRIBUTES_TO_IGNORE);
+        checkGenomicsDBAgainstExpected(workspace, INTERVAL, COMBINED, b38_reference_20_21, true, ATTRIBUTES_TO_IGNORE, false, false, true);
+    }
+
+    @DataProvider
+    public Object[][] dataForTestExplicitIndicesInSampleNameMapInTheCloud() {
+        final String GVCFS_WITH_INDICES_BUCKET = "gs://hellbender/test/resources/org/broadinstitute/hellbender/tools/genomicsdb/gvcfs_with_indices/";
```

Comment:
These paths ideally would all be built using `BaseTest.getGCPTestInputPath` which in theory makes it possible to reproduce our test data on your own system if you want to, although good luck to anyone who tries...